finetuner/tuner/callback/training_model_checkpoint.py

```diff
@@ -0,0 +1,137 @@
+import os
+from typing import TYPE_CHECKING, Optional
+
+import numpy as np
+from jina.logging.logger import JinaLogger
+
+from finetuner.helper import get_framework
+from .base import BaseCallback
+
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class TrainingModelCheckpoint(BaseCallback):
```

Comment:
We should rename to just `TrainingModelCheckpoint`, as it saves the whole training state, not just the model

Reply:
you mean TrainingCheckpoint ?